### PR TITLE
Eliminate AnnotationRegistry, rely on autoloading only

### PR DESCRIPTION
**Pull request by Majkl578**
Dropping AnnotationRegistry completely, relying on native autoloadin instead.

Incorporates #61 as a separate commit to keep attributions of @KingCrunch.

### CODE REVIEW

File: tests/Doctrine/Tests/TestInit.php [hunk 1]
```diff
@@ -18,7 +18,3 @@
 });
 
 require_once __DIR__ . '/../../../vendor/autoload.php';
```
Comment (Ocramius) [inline tests/Doctrine/Tests/TestInit.php]: This file is most likely no longer needed

File: lib/Doctrine/Annotations/DocParser.php [hunk 2]
```diff
@@ -443,13 +442,8 @@ private function classExists($fqcn)
...
         }
 
-        // first check if the class already exists, maybe loaded through another AnnotationReader
-        if (class_exists($fqcn, false)) {
-            return $this->classExists[$fqcn] = true;
-        }
-
         // final check, does this class exist?
-        return $this->classExists[$fqcn] = AnnotationRegistry::loadAnnotationClass($fqcn);
+        return $this->classExists[$fqcn] = class_exists($fqcn);
```
Comment (Ocramius) [inline lib/Doctrine/Annotations/DocParser.php]: The entire private method can be dropped

Comment (Ocramius) [reply lib/Doctrine/Annotations/DocParser.php]: If not: what is our reason for caching a failure?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: tests/Doctrine/Tests/TestInit.php (hunk 1 above)
Comment (Majkl578) [reply tests/Doctrine/Tests/TestInit.php]: yup :+1:

File: lib/Doctrine/Annotations/DocParser.php (hunk 2 above)
Comment (Majkl578) [reply lib/Doctrine/Annotations/DocParser.php]: This whole thing will be gone soon once Hoa parser is in place so I'd ignore that for now.

Comment (Ocramius) [reply lib/Doctrine/Annotations/DocParser.php]: Alright, let's not touch it for now 👌

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @KingCrunch and @Majkl578!

File: lib/Doctrine/Annotations/DocParser.php (hunk 2 above)
Comment (Slamdunk) [reply lib/Doctrine/Annotations/DocParser.php]: > This whole thing will be gone soon once Hoa parser is in place

What's that?

Comment (Majkl578) [reply lib/Doctrine/Annotations/DocParser.php]: Surprise. :gift:

**Pull request merged**